vllm/v1/core/block_pool.py

```diff
@@ -130,13 +130,24 @@ def cache_full_blocks(
         assert len(block_hashes) >= num_cached_blocks
         new_block_hashes = block_hashes[num_cached_blocks:]
 
+        regenerate_hash = False
         # Update the new blocks with the block hashes through the chain.
         if num_cached_blocks == 0:
             prev_block_hash_value = None
         else:
             prev_block = blocks[num_cached_blocks - 1]
-            assert prev_block.block_hash is not None
-            prev_block_hash_value = prev_block.block_hash.get_hash_value()
+            if prev_block.block_hash is None:
+                # This will happen for the chunked local attention,
+                # when the new block starts from the begining of a chunk,
+                # previous block is not cached but as computed (null block)
+                # We need to generate the hash key from beginning tokens
+                # (e.g. window of 4, block of 2,
+                # [0,1], [2, 3] [4,5] will be null, null, new block),
+                # the hash key will be re-generated as of hash(0,1,2,3,4,5)
+                prev_block_hash_value = None
+                regenerate_hash = True
+            else:
+                prev_block_hash_value = prev_block.block_hash.get_hash_value()
```

Comment:
Feel that `regenerate_hash` is a bit hacky. Can you initialize prev_block_hash_value from the block_hashes list?